Core/Services/StitchCoreRemoteMongoDBService/Tests/StitchCoreRemoteMongoDBServiceTests/Sync/DataSynchronizerUnitTests.swift

```diff
@@ -114,4 +117,84 @@ class DataSynchronizerUnitTests: XCMongoMobileTestCase {
 
         XCTAssertFalse(dataSynchronizer.isRunning)
     }
+
+    func testCount() throws {
+        dataSynchronizer.configure(namespace: namespace,
+                                   conflictHandler: TestConflictHandler(),
+                                   changeEventListener: TestEventListener(),
+                                   errorListener: TestErrorListener())
+        XCTAssertEqual(0, try dataSynchronizer.count(in: namespace))
+
+        let doc1 = ["hello": "world", "a": "b"] as Document
+        let doc2 = ["hello": "computer", "a": "b"] as Document
+
+        try XCMongoMobileTestCase.client.db(DataSynchronizer.localUserDBName(withInstanceKey: instanceKey.oid, for: namespace))
+            .collection(namespace.collectionName, withType: Document.self).insertMany([doc1, doc2])
+
+        XCTAssertEqual(2, try dataSynchronizer.count(in: namespace))
+
+        try XCMongoMobileTestCase.client.db(DataSynchronizer.localUserDBName(withInstanceKey: instanceKey.oid, for: namespace))
+            .collection(namespace.collectionName, withType: Document.self).deleteMany(Document())
+
+        XCTAssertEqual(0, try dataSynchronizer.count(in: namespace))
+    }
+    
+    func testFind() throws {
+        dataSynchronizer.configure(namespace: namespace,
+                                   conflictHandler: TestConflictHandler(),
+                                   changeEventListener: TestEventListener(),
+                                   errorListener: TestErrorListener())
+        XCTAssertEqual(0, try dataSynchronizer.count(in: namespace))
+
+        let doc1 = ["hello": "world", "a": "b"] as Document
+        let doc2 = ["hello": "computer", "a": "b"] as Document
+
+        try XCMongoMobileTestCase.client.db(DataSynchronizer.localUserDBName(withInstanceKey: instanceKey.oid, for: namespace))
+            .collection(namespace.collectionName, withType: Document.self).insertMany([doc1, doc2])
+
+        let cursor: MongoCursor<Document> =
+            try dataSynchronizer.find(filter: ["hello": "computer"], options: nil, in: namespace)
+
+        XCTAssertEqual(2, try dataSynchronizer.count(in: namespace))
+
+        let actualDoc = cursor.next()
+
+        XCTAssertEqual("b", actualDoc?["a"] as? String)
```

Comment:
[nit] maybe add something along the lines of `guard let strongActualDoc = actualDoc { XCTFail(...) }` so you don't have to do `actualDoc?[...]` in the assertions

Reply:
I actually think that'll make the test more difficult to read. It'll fail all the same, and it doesn't make sense to guard for something we intend to fail. Plus it will fail in a more appropriate spot.

`XCTAssertEqual("b", actualDoc?["a"] as? String)` would fail to `XCTAssertEqual failed: ("Optional("b")") is not equal to ("nil")` on the line it fails on, as opposed to unnecessarily guarding and failing in a separate location, when it should all be one check.